prov/lnx/include/lnx.h

```diff
@@ -200,10 +200,13 @@ struct lnx_peer {
 struct lnx_peer_table {
 	struct util_av lpt_av;
 	int lpt_max_count;
-	int lpt_count;
 	struct lnx_domain *lpt_domain;
-	/* an array of peer entries */
-	struct lnx_peer **lpt_entries;
+	pthread_rwlock_t lpt_lock;
+	void (*lpt_rlock)(pthread_rwlock_t *l);
+	void (*lpt_wlock)(pthread_rwlock_t *l);
+	void (*lpt_unlock)(pthread_rwlock_t *l);
```

Comment:
I think the fine tuning of the locks here is unnecessary and we can just use the lock in the util_av (which we can turn into a mutex to optimize out). If an application is interested in performant threading, then they should be using separate domains (and thus AVs) for their threads.